pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -268,19 +288,28 @@ func ApplyTaskResultsToPipelineResults(
 						} else {
 							arrayReplacements[v1beta1.StripStarVarSubExpression(variable)] = resultValue.ArrayVal
 						}
+					case v1beta1.ParamTypeObject:
+						objectReplacements[v1beta1.StripStarVarSubExpression(variable)] = resultValue.ObjectVal
```

Comment:
I think the validation to check keys against param definition is that compare the resultValue.ObjectVal with PipelineSpec, which is passed in from function